packages/swing-store-lmdb/test/test-state.js

```diff
@@ -52,15 +52,140 @@ test('storageInLMDB under SES', t => {
   fs.rmdirSync(dbDir, { recursive: true });
   t.is(isSwingStore(dbDir), false);
   const { kvStore, commit, close } = initSwingStore(dbDir);
-  testStorage(t, kvStore);
+  testKVStore(t, kvStore);
   commit();
   const before = getAllState(kvStore);
   close();
   t.is(isSwingStore(dbDir), true);
 
-  const { kvStore: after } = openSwingStore(dbDir);
+  const { kvStore: after, close: close2 } = openSwingStore(dbDir);
   t.deepEqual(getAllState(after), before, 'check state after reread');
   t.is(isSwingStore(dbDir), true);
+  close2();
+});
+
+test('streamStore read/write', t => {
+  const dbDir = 'testdb';
+  t.teardown(() => fs.rmdirSync(dbDir, { recursive: true }));
+  fs.rmdirSync(dbDir, { recursive: true });
+  t.is(isSwingStore(dbDir), false);
+  const { streamStore, commit, close } = initSwingStore(dbDir);
+
+  let s1pos;
+  const writer1 = streamStore.openWriteStream('st1');
+  s1pos = writer1('first', s1pos);
+  s1pos = writer1('second', s1pos);
+  const s1posAlt = { ...s1pos };
+  const writer2 = streamStore.openWriteStream('st2');
+  s1pos = writer1('third', s1pos);
+  let s2pos = { offset: 0, itemCount: 0 };
+  s2pos = writer2('oneth', s2pos);
+  s1pos = writer1('fourth', s1pos);
+  s2pos = writer2('twoth', s2pos);
+  const s2posAlt = { ...s2pos };
+  s2pos = writer2('threeth', s2pos);
+  s2pos = writer2('fourst', s2pos);
+  streamStore.closeStream('st1');
+  streamStore.closeStream('st2');
+  const reader1 = streamStore.openReadStream(
+    'st1',
+    streamStore.STREAM_START,
+    s1pos,
+  );
+  const reads1 = [];
+  for (const item of reader1) {
+    reads1.push(item);
+  }
+  t.deepEqual(reads1, ['first', 'second', 'third', 'fourth']);
+  const writer2alt = streamStore.openWriteStream('st2');
+  s2pos = writer2alt('re3', s2posAlt);
+  streamStore.closeStream('st2');
+  const reader2 = streamStore.openReadStream(
+    'st2',
+    streamStore.STREAM_START,
+    s2pos,
+  );
+  const reads2 = [];
+  for (const item of reader2) {
+    reads2.push(item);
+  }
+  t.deepEqual(reads2, ['oneth', 'twoth', 're3']);
+
+  const reader1alt = streamStore.openReadStream('st1', s1posAlt, s1pos);
+  const reads1alt = [];
+  for (const item of reader1alt) {
+    reads1alt.push(item);
+  }
+  t.deepEqual(reads1alt, ['third', 'fourth']);
+
+  const writerEmpty = streamStore.openWriteStream('empty');
+  const emptyPos = writerEmpty('filler');
+  streamStore.closeStream('empty');
+  const readerEmpty = streamStore.openReadStream('empty', emptyPos, emptyPos);
+  const readsEmpty = [];
+  for (const item of readerEmpty) {
+    readsEmpty.push(item);
+  }
+  t.deepEqual(readsEmpty, []);
+  const readerEmpty2 = streamStore.openReadStream(
+    'empty',
+    streamStore.STREAM_START,
+    streamStore.STREAM_START,
+  );
+  const readsEmpty2 = [];
+  for (const item of readerEmpty2) {
+    readsEmpty2.push(item);
+  }
+  t.deepEqual(readsEmpty2, []);
+
```

Comment:
Could you add one more test (if it isn't already covered by one of those and I'm just not reading it correctly):

* open, write(one), write(two), stash last position, close
* read(stashed_position), assert the results are empty
* open, write(three), write(four), close
* read(stashed_position), assert results are (three, four)

I just want something that matches what VatWarehouse is expecting to do.

Reply:
Tests already cover that case.